<launch>
  <arg name="car1_name" default="car1" />
  <arg name="car1_x" default="-1.0" />
  <arg name="car1_y" default="5.0" />
  <arg name="car1_z" default="0.01" />

  <group ns="$(arg car1_name)">
    <!-- Load the URDF for car1 into the ROS Parameter Server -->
    <param name="robot_description" command="$(find xacro)/xacro '$(find multi_robot_scenario)/xacro/p3dx/pioneer3dx.xacro' "/>
    <param name="laser_frame_id" value="car1/front_laser"/>
    <!-- Run a python script to send a service call to gazebo_ros to spawn a URDF robot for car1 -->
    <node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen" args="-urdf -model $(arg car1_name) -param robot_description -x $(arg car1_x) -y $(arg car1_y) -z $(arg car1_z)"/>

    <!-- Robot State Publisher for car1 -->
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" >
      <param name="publish_frequency" type="double" value="30.0"/>
      <param name="tf_prefix" type="string" value="$(arg car1_name)"/>
    </node>

    <!-- Joint State Publisher for car1 -->
    <node pkg="joint_state_publisher" type="joint_state_publisher" name="joint_state_publisher" >
      <param name="publish_frequency" type="double" value="30.0"/>
    </node>

    <!-- Combined broadcaster for car1 -->
    <!-- <node name="tf_broadcaster" pkg="multi_robot_scenario" type="publish_odom_from_gazebo.py" output="screen">
      <param name="namespace" value="$(arg car1_name)"/> -->
    <node pkg="multi_robot_scenario" name="publish_odom_from_gazebo_para_car1" type="publish_odom_from_gazebo_model_tf.py" output="screen">
      <param name="car_name" value="car1" />
    </node>

  </group>

  <!-- Launch gazebo with the specified world -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find multi_robot_scenario)/launch/xzh.world"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>

  <!-- RViz Configuration -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find multi_robot_scenario)/launch/pioneer3dx.rviz"/>

</launch>

